README.md

```diff
@@ -1 +1,34 @@
 # static-ls
+
+static-ls is a hie and hiedb based language server heavily inspired by halfsp.
+
+The goal of static-ls is to provide a high-speed, low-memory solution for large
+projects which make extensive use of template Haskell (and thus where
+haskell-language-server tends to take up too much memory on recompilation).
+Haskell-language-server is recommended if you are not experiencing these
+issues. `static-ls` is meant to work on enterprise size projects.
+
+## Usage
+
+1. Compile your project with ide info `-fwrite-ide-info` and `-hiedir .hiefiles`
+  - If you're using hpack you can add:
+    ```
+      ghc-options:
+        - -fwrite-ide-info
+        - -hiedir .hiefiles
```

Comment:
i want to note that it might be simpler to not use the `-hiedir` option.  you can still support it, but if `static-ls` were to work with out it then it's a simpler user experience to not have to specify the `-hiedir` option

Reply:
I think for the first iteration of this I'll leave it with the `hiedir` requirement - but remove it in a future PR

I definitely do want to resolve the build/hie directory by default as well as accept this as an argument (or make this configurable)